packages/flutter/lib/src/painting/box_decoration.dart

```diff
@@ -213,18 +212,18 @@ class BoxDecoration extends Decoration {
   final BoxShape shape;
 
   @override
-  EdgeInsetsGeometry get padding => border?.dimensions;
+  EdgeInsetsGeometry? get padding => border?.dimensions;
 
   @override
-  Path getClipPath(Rect rect, TextDirection textDirection) {
-    Path clipPath;
+  Path? getClipPath(Rect rect, TextDirection textDirection) {
```

Comment:
This is surprising, shape is never null, so this should always return non-null.

Reply:
null can be returned with BoxShape.rectangle and a null borderRadius.